### PR TITLE
Integrate migration cancel feature

### LINKED ISSUE #425: Add ability to cancel a Stage/Migrate in progress
Backend issue to support:  https://github.com/fusor/mig-controller/issues/237

### PR DESCRIPTION
**Pull request by Danil-Grigorev**
cancel-spelling-fix

Spelling fixes are in the https://github.com/konveyor/mig-controller/pull/476

Closes #425, depends on #792

### CODE REVIEW

**Comment by Danil-Grigorev** [comment]
@vconzola What else is needed to be done from UX perspective?

**Comment by Danil-Grigorev** [comment]
@eriknelson @ibolton336  Cancelling a final migration does not prevent UI from setting `Closed` on the migplan. It is blocking the PR from being complete, assuming UX standpoint is ok.

**Comment by Danil-Grigorev** [comment]
@eriknelson  Now the design is following mockups. Please review

**Comment by eriknelson** [comment]
@Danil-Grigorev linter is failing test

**Comment by Danil-Grigorev** [comment]
> @Danil-Grigorev linter is failing test

Fixed. Ready for review after @vconzola approval.

**Comment by vconzola** [comment]
@Danil-Grigorev There are still a number of places where the implementation doesn't match the design.
1. Spelling of "canceling" and "canceled" should be consistent with only one "l".
2. It doesn't look like the PF dropdown with kebab component was used for the kebab action menu.
3. The "Last state" of the migration plan should remain "Running" until the cancel is confirmed to have succeeded.
4. When the cancel is confirmed to have succeeded the "Last state" of the migration plan should change to "Migration canceled" with no icon. If the cancel fails the state should remain "Running".
5. The toast notification should not be displayed until the cancel is confirmed to have succeeded or failed. It should be located in the upper right corner (as shown), not centered. It should include the appropriate icon (success/fail) and a close X to dismiss it. It doesn't look like this was done using the PF component.
6. As soon as the cancel request is sent to the backend the stage/migration status should change to "Canceling stage" or "Canceling migration" and should remain in that status until the cancel confirmation is received. At that point the status should change to "Canceled".

**Comment by Danil-Grigorev** [comment]
@vconzola @eriknelson

**Comment by eriknelson** [comment]
@Danil-Grigorev :+1: will review probably on monday

File: src/app/home/components/DataList/Plans/MigrationsTable.tsx [hunk 1]
```diff
@@ -84,7 +94,7 @@ const MigrationsTable: React.FunctionComponent<IProps> = ({ migrations, isPlanLo
           TODO: ** need to evaluate what text to show here **
           </Title> */}
         </EmptyState>
-      </Bullseye>
+      </Bullseye >
```
Comment (eriknelson) [inline src/app/home/components/DataList/Plans/MigrationsTable.tsx:97]: Wondering why this isn't failing a lint rule? There shouldn't be whitespace here.

File: src/app/home/components/DataList/Plans/MigrationsTable.tsx [hunk 2]
```diff
@@ -104,12 +114,13 @@ const MigrationsTable: React.FunctionComponent<IProps> = ({ migrations, isPlanLo
...
           <EmptyState variant="small">
             <Title headingLevel="h2" size="xl">
-                No migrations started
+              No migrations started
             </Title>
           </EmptyState>
         </Bullseye>
       )}
-    </React.Fragment>
+    </React.Fragment >
```
Comment (eriknelson) [inline src/app/home/components/DataList/Plans/MigrationsTable.tsx:122]: Here as well.

**Pull request merged**